pkg/kds/v2/reconcile/reconciler.go

```diff
@@ -71,15 +75,15 @@ func (r *reconciler) Reconcile(ctx context.Context, node *envoy_core.Node) error
 	if new == nil {
 		return errors.New("nil snapshot")
 	}
-	id := r.hasher.ID(node)
+	id := CustomHashId(ctx, r.hasher.ID(node))
```

Comment:
Anyway to have the node decorated with this instead of keeping it in the context? I feel like it'd containt complexity a bit more.